### ISSUE TITLE
False positive for array iterator when break is used

### ISSUE THREAD

**Issue by kubawerlos** [vimeo/psalm#2546]
For a code:
```php
<?php

$numbers = [];
for ($i = 0; $i < 1000; $i++) {
    if (count($numbers) < 20) {
        $numbers[] = $i;
        continue;
    }
    echo 'Enough';
    break;
}
```
the output is:
```
INFO: UnusedVariable - 4:25 - Variable $i is never referenced
```
https://psalm.dev/r/34aa466b06

In the real-life case I have few ifs in the loop (some even having `$i` as argument) and each ends with `continue` and when none is entered I want the loop to end.

Removing the `break` makes the error goes away (together with few passing tests).

**Comment by muglug** [comment]
Yeah, I think `for` vars should always be exempted.

Also this code should have no bugs:

```php
$a = 0;
for ($i = 0; $i < 1000; $i++) {
    if ($a < 20) {
        $a += 2;
        echo "hello";
        continue;
    }
    echo 'Enough';
    break;
}
```

**Comment by muglug** [comment]
Didn't actually fix original issue
